core/src/main/java/com/linecorp/armeria/client/HttpClientBuilder.java

```diff
@@ -38,13 +38,13 @@
     /**
      * An undefined {@link URI} to create {@link HttpClient} without specifying {@link URI}.
      */
-    private static final URI UNDEFINED_URI = URI.create("none+http://undefined");
+    private static final URI UNDEFINED_URI = URI.create("http://undefined");
```

Comment:
Revert?

Reply:
See above comment regarding isUndefinedUri change

Reply:
I have made changes to handle UNDEFINED_URI, but haven't reverted the value of constant. It's because, per this PR - we don't want uri() to return _none+_. 
I think, it would not be idiomatic, to have _none+_ returned for this special case. We are doing an == check for this constant though.